src/sentry/static/sentry/app/views/workBatchDetailsWaitingToBeMerged/workbatchDetails.jsx

```diff
@@ -4,102 +4,153 @@ import withOrganization from 'app/utils/withOrganization';
     const arr = Object.values(byIds);
     let wipWorkbatch = arr[0];
-    this.props.getConfiguredStaticContents(
+    return this.props.getConfiguredStaticContents(
       this.props.organization,
       wipWorkbatch.cls_full_name
     );
   }
 
-  render() {
-    //TODO: merge this with files in the workBatchDetails folder
-    const buttonsFromConfig = [];
+  fetchDetailedContentFromWip() {
+    const byIds = this.props.workBatchEntry.byIds;
+    const arr = Object.values(byIds);
+    let wipWorkbatch = arr[0];
+    return this.props.getWorkBatchDetails(this.props.organization, wipWorkbatch.id);
+  }
+
+  fetchDefinitionsFromProps() {
     let workBatchDefinitionEntry = this.props.workBatchDefinitionEntry;
     if (workBatchDefinitionEntry == null) {
-      return <LoadingIndicator />;
+      return null;
     }
     let {byIds, detailsId} = workBatchDefinitionEntry;
     if (detailsId == null) {
+      return null;
+    }
+    let workDefinition = byIds[detailsId];
+    let {fields, buttons, id} = workDefinition;
+    if (fields === null || buttons === null || id === null) {
+      return null;
+    }
+    return workDefinition;
+  }
+
+  sendButtonClickedEvent(buttonEvent) {
+    this.props.sendButtonClickedEvent(this.props.organization, buttonEvent);
+  }
+
+  render() {
+    //TODO: merge this with files in the workBatchDetails folder
+    if (
+      this.props.workBatchDetailsEntry.loadingDetails ||
+      this.state === null ||
+      !this.state.fetched
+    ) {
       return <LoadingIndicator />;
     }
-    const workDefinition = byIds[detailsId];
-    let {buttons: buttonDefinitions, id: configId} = workDefinition;
-    for (const entry of buttonDefinitions) {
-      const buttonClick = () => {
-        this.sendButtonClickedEvent(entry.name, configId);
-      };
-      buttonsFromConfig.push(
-        <button
-          className="btn btn-sm btn-default"
-          onClick={buttonClick}
-          name={entry.name}
-        >
-          {entry.caption}
-        </button>
-      );
+    let workDefinition = this.fetchDefinitionsFromProps();
+    if (workDefinition === null) {
+      return <LoadingIndicator />;
     }
+    let {detailsId: workbatchId} = this.props.workBatchDetailsEntry;
     return (
-      <div>
-        <h1>Workbatch details proxy</h1>
-        {buttonsFromConfig}
-      </div>
+      <DetailsForm
+        workDefinition={workDefinition}
+        sendButtonClickedEvent={this.sendButtonClickedEvent}
+        handleChange={this.handleChange}
+        currentFieldValues={this.state.currentFieldValues}
+        workBatchId={workbatchId}
+      />
     );
   }
 
-  sendButtonClickedEvent(clickedButtonName, workBatchId) {
-    const buttonEvent = {
-      event: clickedButtonName,
-      workBatchId,
-    };
-    this.props.sendButtonClickedEvent(this.props.organization, buttonEvent);
+  initCurrentFieldValues() {
+    let {detailsId, byIds} = this.props.workBatchDetailsEntry;
+    let workbatch = byIds[detailsId];
```

Comment:
Is there some kind of check to ensure this entry exists in ByIds?

Reply:
I'll add it.